### PR TITLE
Add BalanceStatus for insufficient wordings

### PR DESCRIPTION
**Pull request by hewigovens**
Add wordings for these cases:

- insufficientEther
- insufficientGas
- insufficientToken

### CODE REVIEW

File: Trust/Transfer/Types/BalanceStatus.swift [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+            if !gasSufficient {
+                return .insufficientGas
+            }
+        }
+        return .none
+    }
+
+    var insufficientText: String {
+        let key = insufficientTextKey
+        if key == Key.insufficientEther {
```
Comment (vikmeup) [inline Trust/Transfer/Types/BalanceStatus.swift]: can we do switch statement in here instead?

File: Trust/Transfer/Types/BalanceStatus.swift [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+                return "send.error.insufficientGas"
+            case .insufficientToken:
+                return "send.error.insufficientToken"
+            case .none:
+                return ""
+            }
+        }
+    }
+
+    private func _L(_ key: String, value: String? = .none) -> String {
```
Comment (vikmeup) [inline Trust/Transfer/Types/BalanceStatus.swift]: Can we avoid `_L` method? make it hard to read. Using `NSLocalizedString ` should be ok

File: Trust/Transfer/Types/BalanceStatus.swift [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+        return NSLocalizedString(key, value: value ?? key, comment: "")
+    }
+
+    var sufficient: Bool {
```
Comment (vikmeup) [inline Trust/Transfer/Types/BalanceStatus.swift]: I would have separate test unit test just for `BalanceStatus` as well

**Comment by vikmeup** [comment]
@hewigovens Did you consider having `BalanceStatus` as keys itself? 
```
enum BalanceStatus {
     case ether(etherSufficient: Bool, gasSufficient: Bool)
     case token(tokenSufficient: Bool, gasSufficient: Bool)
 }
```
=> 
```
enum BalanceStatus {
    case insufficientEther
    case insufficientGas
    case insufficientToken
    case correct
}
```

**Comment by hewigovens** [comment]
@vikmeup If we make BalanceStatus flat, it will lose the context for insufficientGas

**Comment by OlegGordiichuk** [comment]
Nice work.

**Pull request merged**